File: example/src/test/java/ru/virgil/example/util/security/MockAuthorizationService.java
package ru.virgil.example.util.security;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import ru.virgil.example.security.SecurityUserDetails;
import ru.virgil.example.security.SecurityUserDetailsService;
import ru.virgil.security.firebase.FirebaseAuthenticationToken;

@Service
@RequiredArgsConstructor
public class MockAuthorizationService implements ru.virgil.security.mock.MockAuthorizationService {

    private final SecurityUserDetailsService securityUserDetailsService;

    @Override
    public SecurityUserDetails authOrRegister(FirebaseAuthenticationToken token) {
        return securityUserDetailsService.getRepository().findByFirebaseUserId(token.getPrincipal().toString())
                .orElseGet(() -> securityUserDetailsService.register(token));
    }
}

File: example/src/test/java/ru/virgil/example/util/security/MockAuthorizationService.kt
package ru.virgil.example.util.security

import org.springframework.stereotype.Service
import ru.virgil.example.security.SecurityUserDetails
import ru.virgil.example.security.SecurityUserDetailsService
import ru.virgil.security.firebase.FirebaseAuthenticationToken
import ru.virgil.security.mock.MockAuthorizationService

@Service
class MockAuthorizationService(
    private val securityUserDetailsService: SecurityUserDetailsService,
) : MockAuthorizationService {

    override fun authOrRegister(token: FirebaseAuthenticationToken): SecurityUserDetails {
        return securityUserDetailsService.repository.findByFirebaseUserId(token.principal.toString())
            .orElseGet { securityUserDetailsService.register(token) }
    }
}
